src/marqo/index.py

```diff
@@ -165,19 +170,29 @@ def create(config: Config, index_name: str,
             cl_ix_defaults = cl_settings['index_defaults']
             cl_ix_defaults['treat_urls_and_pointers_as_images'] = \
                 create_index_settings.treat_urls_and_pointers_as_images
-            if model is not None:
+            if create_index_settings.model is not None:
                 cl_ix_defaults['model'] = create_index_settings.model
+            else:
+                cl_ix_defaults.pop('model', None)
             cl_ix_defaults['normalize_embeddings'] = create_index_settings.normalize_embeddings
             cl_text_preprocessing = cl_ix_defaults['text_preprocessing']
             cl_text_preprocessing['split_overlap'] = create_index_settings.sentence_overlap
             cl_text_preprocessing['split_length'] = create_index_settings.sentences_per_chunk
             cl_img_preprocessing = cl_ix_defaults['image_preprocessing']
             if image_preprocessing_method is not None:
                 cl_img_preprocessing['patch_method'] = create_index_settings.image_preprocessing_method
+            else:
+                cl_img_preprocessing.pop('patch_method', None)
             if not config.is_marqo_cloud:
                 return req.post(f"indexes/{index_name}", body=cl_settings)
-            cl_settings['inference_type'] = create_index_settings.inference_type
-            cl_settings['storage_class'] = create_index_settings.storage_class
+            if create_index_settings.inference_type is not None:
+                cl_settings['inference_type'] = create_index_settings.inference_type
```

Comment:
do we need logic to ensure we set this to either `inference_type` XOR `inference_node_type`? same goes for storage and counts

Reply:
We may need to add new methods to CreateIndexSettings that gets the correct attribute

Reply:
during index config creation if deprecated param is passed and None is passed for it's new param we simply state that new parameter equals passed deprecated value and we never set the deprecated parameter